### PR TITLE
Js/test coverage

### PR DESCRIPTION
**Pull request by mutexkid**
In this pull request, i make a first pass at adding test coverage to alexa-app. 
- add tests for current card api
- add tests for intent handling behavior
- add tests for session behavior
- add tests for initialization

I also have add travis and code coverage integration so that we've got a way to check new pull requests pass the test suite. 

@matt-kruse : For the travis integration to be active on the project, please enable the travis integration on the github repository - I currently don't have access to enable that in settings. 
https://www.evernote.com/l/ATn8zc-KNIFOsLbcH5p9m-70PqIwgHUTIp4B/image.png
Also, the code coverage report integration is via coveralls. 
Here is the output: https://coveralls.io/builds/6062172

### CODE REVIEW

**Comment by mutexkid** [comment]
@radishmouse ^^

**Comment by mutexkid** [comment]
@rickwargo

File: test/test_alexa_app_launch_request.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    .reprompt(expectedReprompt)
+                    .reprompt(expectedReprompt);
+                });
+                var subject = app.request(mockRequest).then(function(response) {
+                  return response.response.reprompt.outputSpeech;
+                });
+                return expect(subject).to.eventually.become({
+                  ssml: "<speak>" + expectedReprompt + "</speak>",
+                  type: "SSML"
+                });
```
Comment (mutexkid) [inline test/test_alexa_app_launch_request.js]: this result also seemed strange - i saw in the code path logic for combining reprompts but i didn't observe that behavior.

**Comment by matt-kruse** [comment]
Added Travis CI

**Pull request merged**